### ISSUE TITLE
problem with regular expressions inside macros

### ISSUE THREAD

**Issue by francescoagati** [massiveinteractive/haxe-react#2]
This, 
isn't a real bug of haxe-react but a bug of some distributions of haxe. Some cases of regular expression in haxe aren't supported.
In my case i have found this compiler error on my computer

```
Main.hx:11: lines 11-19 : Invalid JSX: Unsupported escaped char '{'
```

**Comment by elsassph** [comment]
Our parser is a bit naive for now - it should be rewritten. For now it is recommended to avoid complex expressions.

**Comment by impaler** [comment]
I get the same error trying to compile the example on ubuntu using the shiny haxe ppa, its definitely a roadblock for some.

@francescoagati did you find a solution? I haven't encountered this issue with Regex engines before but I was only really using haxe on mac or windows (I don't want to return to windows).

@elsassph are you planning or have you started this rewrite you mention? I can't really take your advice to avoid complex expressions when the only example around wont compile.

This project is awesome btw, despite the jsx "compromise".

**Comment by elsassph** [comment]
Do you mean the todo-app example didn't compile?

We didn't start the rewrite unfortunately. The main project using the lib right now had decided to go without jsx.

**Comment by impaler** [comment]
@elsassph yeah the todo example threw this same error when building on ubuntu 15.10 with the haxe ppa.

**Comment by bablukid** [comment]
I have the same problem. Strangely I can successfully compile my project on windows, but 2 other developers cant compile the same project on Linux ( with the same version of haxe , same version of lib ).

**Comment by elsassph** [comment]
It should be fixed in the new 1.0.0 release - but it may require some work to update your project.
